core/src/types/hash.rs

```diff
@@ -146,14 +146,17 @@ impl From<Hash> for crate::tendermint::Hash {
     }
 }
 
-#[cfg(test)]
-mod tests {
+#[cfg(any(test, feature = "testing"))]
+/// Tests and strategies for hashes
+pub mod tests {
     use proptest::prelude::*;
+    #[cfg(test)]
     use proptest::string::{string_regex, RegexGeneratorStrategy};
 
     use super::*;
 
     /// Returns a proptest strategy that yields hex encoded hashes.
+    #[cfg(test)]
```

Comment:
odd looking hack, since the module is already flagged as a test module

Reply:
I've refactored the testing modules a bit in here [6936ac2](https://github.com/anoma/namada/pull/2259/commits/6936ac2c3c564ac55eb6c45256fb313068752851) - the strategies and test helpers should be separate sub-module (`mod testing`) and actual tests in `mod test(s)`

Reply:
so that "testing" can be re-used in crates that depend on these